ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member

```
java.lang.AssertionError: Expecting only 1 clear event expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest$29.run(ClientMapNearCacheTest.java:1053)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:967)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:984)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member(ClientMapNearCacheTest.java:1050)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-freestyle/134/testReport/junit/com.hazelcast.client.map.impl.nearcache/ClientMapNearCacheTest/receives_one_clearEvent_after_mapEvictAll_call_from_member_batchInvalidationEnabled_false_/

possibly related to #9940

```
java.lang.AssertionError: Expecting only 1 clear event expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest$29.run(ClientMapNearCacheTest.java:1053)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:967)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:984)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member(ClientMapNearCacheTest.java:1050)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-freestyle/164/testReport/junit/com.hazelcast.client.map.impl.nearcache/ClientMapNearCacheTest/receives_one_clearEvent_after_mapEvictAll_call_from_member_batchInvalidationEnabled_true_/

```
java.lang.AssertionError: Expecting only 1 clear event expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest$29.run(ClientMapNearCacheTest.java:1053)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:967)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:984)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member(ClientMapNearCacheTest.java:1050)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-ZuluJDK6-freestyle/43/testReport/junit/com.hazelcast.client.map.impl.nearcache/ClientMapNearCacheTest/receives_one_clearEvent_after_mapEvictAll_call_from_member_batchInvalidationEnabled_false_/

**maintenance**
```
java.lang.AssertionError: Expecting only 1 clear event expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest$29.run(ClientMapNearCacheTest.java:1053)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:967)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:984)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member(ClientMapNearCacheTest.java:1050)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-freestyle/61/testReport/junit/com.hazelcast.client.map.impl.nearcache/ClientMapNearCacheTest/receives_one_clearEvent_after_mapEvictAll_call_from_member_batchInvalidationEnabled_true_/

*master*
```
java.lang.AssertionError: Expecting only 1 clear event expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest$27.run(ClientMapNearCacheTest.java:1009)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1003)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1020)
	at com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest.receives_one_clearEvent_after_mapEvictAll_call_from_member(ClientMapNearCacheTest.java:1006)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-ZuluJDK7/com.hazelcast$hazelcast-client/565/testReport/junit/com.hazelcast.client.map.impl.nearcache/ClientMapNearCacheTest/receives_one_clearEvent_after_mapEvictAll_call_from_member_batchInvalidationEnabled_false_/

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
@@ -84,9 +84,6 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
 
     protected final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
 
-    private static final Random rand = new Random();
-    private static final int NIGHTLY_TEST_WAIT_DURATION = 30;
-
     @After
     public void tearDown() {
         hazelcastFactory.shutdownAll();
@@ -911,7 +908,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, getRandomTime());
+        }, 10);
     }
 
     @Test
@@ -962,7 +959,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, getRandomTime());
+        }, 10);
     }
 
     private void mapClearFromMember(final ClearEventCounterEventHandler handler) {
@@ -1006,7 +1003,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, getRandomTime());
+        }, 10);
     }
 
     private void mapEvictAllFromClient(final ClearEventCounterEventHandler handler) {
@@ -1050,7 +1047,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, getRandomTime());
+        }, 10);
     }
 
     private void mapAvictAllFromMember(final ClearEventCounterEventHandler handler) {
@@ -1343,13 +1340,5 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         int getClearEventCount() {
             return clearEventCount.get();
         }
-    }
-
-    private static int getRandomTime() {
-        int duration = rand.nextInt(NIGHTLY_TEST_WAIT_DURATION);
-        if (duration == 0) {
-            return 10;
-        }
-        return duration;
     }
 }

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
@@ -39,6 +39,7 @@ import com.hazelcast.nio.serialization.Data;
 import com.hazelcast.spi.properties.GroupProperty;
 import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import com.hazelcast.test.annotation.NightlyTest;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.junit.After;
@@ -82,6 +83,9 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
     }
 
     protected final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+
+    private static final Random rand = new Random();
+    private static final int NIGHTLY_TEST_WAIT_DURATION = 30;
 
     @After
     public void tearDown() {
@@ -896,14 +900,34 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
     }
 
     @Test
+    @Category(NightlyTest.class)
+    public void ensure_receives_one_clearEvent_after_mapClear_call_from_client() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapClearFromClient(handler);
+
+        assertTrueAllTheTime(new AssertTask() {
+            @Override
+            public void run() {
+                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
+            }
+        }, getRandomTime());
+    }
+
+    @Test
     public void receives_one_clearEvent_after_mapClear_call_from_client() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapClearFromClient(handler);
+    }
+
+    private void mapClearFromClient(final ClearEventCounterEventHandler handler) {
         // populate Near Cache
         IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
         // add test listener to count clear events
-        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // create a new client to send events
@@ -917,24 +941,37 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
                 assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
+    }
+
+    @Test
+    public void receives_one_clearEvent_after_mapClear_call_from_member() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapClearFromMember(handler);
+    }
+
+    @Test
+    @Category(NightlyTest.class)
+    public void ensure_receives_one_clearEvent_after_mapClear_call_from_member() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapClearFromMember(handler);
 
         assertTrueAllTheTime(new AssertTask() {
             @Override
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, 2);
+        }, getRandomTime());
     }
 
-    @Test
-    public void receives_one_clearEvent_after_mapClear_call_from_member() {
+    private void mapClearFromMember(final ClearEventCounterEventHandler handler) {
         // populate Near Cache
         IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
         // add test listener to count clear events
-        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // clear map from member side
@@ -948,24 +985,37 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
                 assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
+    }
+
+    @Test
+    public void receives_one_clearEvent_after_mapEvictAll_call_from_client() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapEvictAllFromClient(handler);
+    }
+
+    @Test
+    @Category(NightlyTest.class)
+    public void ensure_receives_one_clearEvent_after_mapEvictAll_call_from_client() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapEvictAllFromClient(handler);
 
         assertTrueAllTheTime(new AssertTask() {
             @Override
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, 2);
+        }, getRandomTime());
     }
 
-    @Test
-    public void receives_one_clearEvent_after_mapEvictAll_call_from_client() {
+    private void mapEvictAllFromClient(final ClearEventCounterEventHandler handler) {
         // populate Near Cache
         IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig());
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
         // add test listener to count clear events
-        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // call evictAll
@@ -979,17 +1029,31 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
                 assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
+    }
+
+    @Test
+    public void receives_one_clearEvent_after_mapEvictAll_call_from_member() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapAvictAllFromMember(handler);
+    }
+
+    @Test
+    @Category(NightlyTest.class)
+    public void ensure_receives_one_clearEvent_after_mapEvictAll_call_from_member() {
+        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
+
+        mapAvictAllFromMember(handler);
 
         assertTrueAllTheTime(new AssertTask() {
             @Override
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        }, 2);
+        }, getRandomTime());
     }
 
-    @Test
-    public void receives_one_clearEvent_after_mapEvictAll_call_from_member() {
+    private void mapAvictAllFromMember(final ClearEventCounterEventHandler handler) {
         IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
 
         // populate Near Cache
@@ -997,7 +1061,6 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         populateNearCache(clientMap, 1000);
 
         // add test listener to count clear events
-        final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // call evictAll
@@ -1011,13 +1074,6 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
                 assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
-
-        assertTrueAllTheTime(new AssertTask() {
-            @Override
-            public void run() {
-                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
-            }
-        }, 2);
     }
 
     @Test
@@ -1287,5 +1343,9 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         int getClearEventCount() {
             return clearEventCount.get();
         }
+    }
+
+    private static int getRandomTime() {
+        return rand.nextInt(NIGHTLY_TEST_WAIT_DURATION);
     }
 }

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
@@ -1346,6 +1346,10 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
     }
 
     private static int getRandomTime() {
-        return rand.nextInt(NIGHTLY_TEST_WAIT_DURATION);
+        int duration = rand.nextInt(NIGHTLY_TEST_WAIT_DURATION);
+        if (duration == 0) {
+            return 10;
+        }
+        return duration;
     }
 }

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java
@@ -898,7 +898,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
     @Test
     public void receives_one_clearEvent_after_mapClear_call_from_client() {
         // populate Near Cache
-        IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig());
+        IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
@@ -911,30 +911,26 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         IMap<Object, Object> anotherClientMap = anotherClient.getMap(clientMap.getName());
         anotherClientMap.clear();
 
-        // sleep for a while to see there is another clear event coming
-        sleepSeconds(2);
-
         assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
+            }
+        });
+
+        assertTrueAllTheTime(new AssertTask() {
             @Override
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        });
+        }, 2);
     }
 
     @Test
     public void receives_one_clearEvent_after_mapClear_call_from_member() {
-        // start new member and populate map
-        HazelcastInstance member = hazelcastFactory.newHazelcastInstance(newConfig());
-        IMap<Integer, Integer> memberMap = member.getMap("test");
-        populateMap(memberMap, 1000);
-
-        // populate client Near Cache
-        NearCacheConfig nearCacheConfig = newNearCacheConfig();
-        ClientConfig clientConfig = newClientConfig();
-        clientConfig.addNearCacheConfig(nearCacheConfig);
-        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
-        IMap<Integer, Integer> clientMap = client.getMap("test");
+        // populate Near Cache
+        IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
+        populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
         // add test listener to count clear events
@@ -942,17 +938,23 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // clear map from member side
+        HazelcastInstance member = hazelcastFactory.getAllHazelcastInstances().iterator().next();
+        IMap memberMap = member.getMap(clientMap.getName());
         memberMap.clear();
-
-        // sleep for a while to see there is another clear event coming
-        sleepSeconds(2);
 
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
-                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
+                assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
+
+        assertTrueAllTheTime(new AssertTask() {
+            @Override
+            public void run() {
+                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
+            }
+        }, 2);
     }
 
     @Test
@@ -971,44 +973,51 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         IMap<Object, Object> anotherClientMap = anotherClient.getMap(clientMap.getName());
         anotherClientMap.evictAll();
 
-        // sleep for a while to see there is another clear event coming
-        sleepSeconds(2);
-
         assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
+            }
+        });
+
+        assertTrueAllTheTime(new AssertTask() {
             @Override
             public void run() {
                 assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
             }
-        });
+        }, 2);
     }
 
     @Test
     public void receives_one_clearEvent_after_mapEvictAll_call_from_member() {
+        IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig(), 2);
+
         // populate Near Cache
-        IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(newNearCacheConfig());
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
-
-        // member comes
-        HazelcastInstance member = hazelcastFactory.newHazelcastInstance(newConfig());
 
         // add test listener to count clear events
         final ClearEventCounterEventHandler handler = new ClearEventCounterEventHandler();
         ((NearCachedClientMapProxy) clientMap).addNearCacheInvalidationListener(handler);
 
         // call evictAll
+        HazelcastInstance member = hazelcastFactory.getAllHazelcastInstances().iterator().next();
         IMap memberMap = member.getMap(clientMap.getName());
         memberMap.evictAll();
-
-        // sleep for a while to see there is another clear event coming
-        sleepSeconds(2);
 
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
-                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
+                assertTrue("Expecting at least 1 clear event", 0 < handler.getClearEventCount());
             }
         });
+
+        assertTrueAllTheTime(new AssertTask() {
+            @Override
+            public void run() {
+                assertEquals("Expecting only 1 clear event", 1, handler.getClearEventCount());
+            }
+        }, 2);
     }
 
     @Test
@@ -1018,7 +1027,7 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         config.getMapConfig("default").getMapStoreConfig().setEnabled(true).setImplementation(new SimpleMapStore());
 
         // populate Near Cache
-        final IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(config, newNearCacheConfig());
+        final IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(config, newNearCacheConfig(), 2);
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
@@ -1044,15 +1053,13 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
         Config config = newConfig();
         config.getMapConfig("default").getMapStoreConfig().setEnabled(true).setImplementation(new SimpleMapStore());
 
-        // member comes
-        HazelcastInstance member = hazelcastFactory.newHazelcastInstance(config);
-
         // populate Near Cache
-        final IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(config, newNearCacheConfig());
+        final IMap<Integer, Integer> clientMap = getNearCachedMapFromClient(config, newNearCacheConfig(), 2);
         populateMap(clientMap, 1000);
         populateNearCache(clientMap, 1000);
 
         // create a new client to send events
+        HazelcastInstance member = hazelcastFactory.getAllHazelcastInstances().iterator().next();
         IMap<Object, Object> memberMap = member.getMap(clientMap.getName());
         memberMap.loadAll(true);
 
@@ -1207,12 +1214,20 @@ public class ClientMapNearCacheTest extends NearCacheTestSupport {
     }
 
     protected <K, V> IMap<K, V> getNearCachedMapFromClient(NearCacheConfig nearCacheConfig) {
-        return getNearCachedMapFromClient(newConfig(), nearCacheConfig);
+        return getNearCachedMapFromClient(newConfig(), nearCacheConfig, 1);
+    }
+
+    protected <K, V> IMap<K, V> getNearCachedMapFromClient(NearCacheConfig nearCacheConfig, int clusterSize) {
+        return getNearCachedMapFromClient(newConfig(), nearCacheConfig, clusterSize);
     }
 
     protected <K, V> IMap<K, V> getNearCachedMapFromClient(Config config, NearCacheConfig nearCacheConfig) {
+        return getNearCachedMapFromClient(config, nearCacheConfig, 1);
+    }
+
+    protected <K, V> IMap<K, V> getNearCachedMapFromClient(Config config, NearCacheConfig nearCacheConfig, int clusterSize) {
         String mapName = randomMapName();
-        hazelcastFactory.newHazelcastInstance(config);
+        hazelcastFactory.newInstances(config, clusterSize);
 
         nearCacheConfig.setName(mapName + "*");
 

